app/controllers/OverviewController.scala

```diff
@@ -110,10 +112,28 @@ class OverviewController @Inject()(
                 messages("site.continue")
               )
             } else {
+              val lastSubmittedPsrFbInPreviousYears =
+                versionsForYears
+                  .filter(x => LocalDate.parse(x.startDate) < yearFrom)
```

Comment:
Would it be better to explicitly search for current tax year - 1  here? In that case the located data would be always from the previous year

Reply:
yeah but what happens if we have a gap year ? That's why I used this logic. I remember we asked this to Josh and he said this is not clear however the most recent submission must be used for the population. lastSubmittedPsrFbInPreviousYears logic must guarantee it.

Reply:
I've asked this question again, let's see if this is more clear now from UX perspective.

Reply:
There is some ongoing discussion about this, we're going to keep this as is for now until that is resolved.
Could you please add a comment @cyeniceri so we can easily identify this later on, in case we need to changed it? E.g. something like `\\ this logic assumes gap years are allowed in pre-population`